p2p/switch.go

```diff
@@ -417,10 +417,10 @@ func (sw *Switch) DialPeersAsync(addrBook AddrBook, peers []string, persistent b
 	for i := 0; i < len(perm); i++ {
 		go func(i int) {
 			j := perm[i]
-
 			addr := netAddrs[j]
-			// do not dial ourselves
-			if addr.Same(ourAddr) {
+
+			if addr.Same(ourAddr) || sw.HasPeerWithAddress(addr) {
+				sw.Logger.Debug("Ignore attempt to connect to an existing peer", "addr", addr)
```

Comment:
Do we want a log message if you try to connect to yourself? If so, would that warrant its own message?

Reply:
Not a biggie since it's `Debug`